Prep for v1.0.0.pre

andremleblanc opened the pull request:
This PR preps for our first release, after we merge this in, I will cut a release to Rubygems.

jcypret commented:
I'm not sure about the gem naming. With the name `helpscout-api`, I would [expect the library's main module](https://guides.rubygems.org/name-your-gem/) to be `Helpscout::API`, but that is not the case here. `Helpscout::API` is the name of the Faraday wrapper class that is part of the library, but it does not wrap the entire library. For instance `Helpscout::Mailbox` and `Helpscout::VERSION` indicate a main module of `Helpscout`. 

<img width="857" alt="Screen Shot 2019-05-28 at 5 01 47 PM" src="https://user-images.githubusercontent.com/1276050/58515405-512e0e00-816a-11e9-9ebd-1b1044e5a5d5.png">

So if the main module is `Helpscout`, using a dash for the second word fine, though maybe we shouldn't have a class constant matching the library name's expected main module?

On a different note, because there's already a library called `helpscout_api`, it's a bit confusing for our version to have such a similar name. @andremleblanc and I brainstormed for a few alternatives and found that `helpscout-sdk` could make sense. That's the terminology HelpScout uses referring to the libraries in their developer docs: https://developer.helpscout.com/mailbox-api/#sdks. I also found a precedence for it in other gems such as `dropbox-sdk` and `aws-sdk`.

<img width="853" alt="Screen Shot 2019-05-28 at 5 14 55 PM" src="https://user-images.githubusercontent.com/1276050/58515989-23e25f80-816c-11e9-958e-a664e658c486.png">

Some other alternative could be `helpscout-wrapper`, `helpscout-ruby`, or `helpscout-v2`.

andremleblanc commented:
**Current Implementation**
Okay, may have went a little overboard with the changes, but if we're going to be semantic, thought I'd go all the way.

- Changed `Helpscout` to `HelpScout` to reflect company naming convention.
- Added "-sdk" to the gem name suffix since that is what this is, a SDK for HelpScout. This RubyGems name is also open. We could also conceivably expand this SDK to support other APIs if HelpScout was to release them.

```
# Gemfile
gem 'help_scout-sdk'

# Local
gem install help_scout-sdk
require 'help_scout_sdk'

# Usage
HelpScout::Mailbox.list
```
**Alternative Implementation**
Only change the gem name to `helpscout-sdk` and keep the "namespace" as `Helpscout`. This would still be a unique name in RubyGems.

```
# Gemfile
gem 'helpscout-sdk'

# Local
gem install helpscout-sdk
require 'helpscout_sdk'

# Usage
Helpscout::Mailbox.list
```

andremleblanc commented:
Zoomed with @jcypret and discussed the change we landed on `help_scout_sdk`.

```
# Gemfile
gem 'help_scout_sdk'

# Local
gem install help_scout_sdk
require 'help_scout'

# Usage
HelpScout::Mailbox.list
```

Ready for review @jcypret @prsimp.

prsimp commented:
Sorry I missed the Zoom. My 2¢ is that it's strange to use the second underscore but drop it from the Ruby constant. I'd expect it to either be hyphenated if its just being included for gem naming reasons (i.e. `gem 'help_scout-sdk', require: 'help_scout'` with `HelpScout::Foo`) or to underscored if we want it in the name (i.e. `gem 'help_scout_sdk'` with `HelpScoutSDK::Foo`).

I hate to see us do more back and forth on this, but I think we should either stick with `-sdk` or include that in our constants. I think if `-api` seems confusing because of the presence of `API`, our best bet is to remove `API` and replace it with `HTTP` or similar.

andremleblanc commented:
Ended up changing the gem name back to `help_scout-sdk` since it feels the most correct even though it looks kinda wonky.
